### PR TITLE
Add section on breaking changes

### PR DESCRIPTION
**Pull request by krlmlr**
to NEWS.

I'm aware of the "Lifecycle changes" section. The new text highlights behavior changes that break tests that worked in dplyr 0.8.4.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
 # dplyr 1.0.0 (in development)
 
+## Breaking changes
+
+* `bind_cols()` no longer forces tibbles.
```
Comment (hadley) [inline NEWS.md]: What do you mean by forces?

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+
+* `bind_rows()` and `combine()` use vctrs coercion rules.
+
+    * For time columns, the time zone of the first argument is used.
+
```
Comment (hadley) [inline NEWS.md]: Need comment about factor + factor and factor + character here

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,30 @@
...
+* `bind_rows()` and `combine()` use vctrs coercion rules.
+
+    * For time columns, the time zone of the first argument is used.
+    
+    * Classed atomic vectors are no longer accepted, `vctrs::vec_is()` is required for all inputs.
+
+* `bind_rows()` and other functions use vctrs name repair, see `?vctrs::vec_as_names`.
+
+* `count()` no longer removes the last level of grouping.
```
Comment (hadley) [inline NEWS.md]: Can you file an issue? I thought `count()` always preserved the previous grouping, so this is a mistake I introduced and we can fix it.

File: NEWS.md [hunk 4]
```diff
@@ -1,5 +1,30 @@
...
+
+* `all.equal.tbl_df()` removed, equality checks for data frames no longer ignore row order or groupings.
```
Comment (hadley) [inline NEWS.md]: Need to comment that this was used by `expect_equal()`; I think it also means that `expect_equal( tibble(x=1), data.frame(x=1))` is now false, but you'd need to check that it was true before.

File: NEWS.md (hunk 3 above)
Comment (krlmlr) [reply NEWS.md]: This was a false negative -- the test was incorrect, which was revealed by the changes in `all.equal.tbl_df()` .

**Comment by hadley** [comment]
Thanks!

**Pull request merged**